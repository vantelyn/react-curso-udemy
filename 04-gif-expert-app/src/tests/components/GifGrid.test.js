import React from 'react';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';
import { GifGrid } from "../../components/GifGrid";
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas en <GifGrid />', () => { 

    const category = 'One Punch';
    
    test('Debe mostrar el snapshot correctamente', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true,
        });

        const wrapper = shallow(<GifGrid category={ category } />);
        expect(wrapper).toMatchSnapshot();
    });

    test('debe mostrar items cuando se cargan imÃ¡genes useFetchGifs', () => {
        
        const gifs = [
            {
            id: 'l3vRlt8kty8KKeHni',
            title: 'one punch GIF by Adult Swim',
            url: "https://media0.giphy.com/media/l3vRlt8kty8KKeHni/giphy.gif?cid=5bcd87eeumlqtq22jd83jf8g5hd086kdi3ko4yz07fxyzsqz&rid=giphy.gif&ct=g",
            },
            {
                id: 'l3vRlt8kty8KKeHhui',
                title: 'one punch GIF by Adult Swim',
                url: "https://media0.giphy.com/media/l3vRlt8kty8KKeHni/giphy.gif?cid=5bcd87eeumlqtq22jd83jf8g5hd086kdi3ko4yz07fxyzsqz&rid=giphy.gif&ct=g",
            },
        ];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false,
        });
        
        const wrapper = shallow(<GifGrid category={ category } />);

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('p').exists() ).toBe( false );
        expect( wrapper.find('GifGridItem').length ).toBe( gifs.length );
        
    });

});