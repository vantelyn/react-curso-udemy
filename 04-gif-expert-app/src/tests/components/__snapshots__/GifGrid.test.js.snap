// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid /> Debe mostrar el snapshot correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    One Punch
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Pruebas en <GifGrid /> debe mostrar items cuando se cargan im√°genes useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    One Punch
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="l3vRlt8kty8KKeHni"
      key="l3vRlt8kty8KKeHni"
      title="one punch GIF by Adult Swim"
      url="https://media0.giphy.com/media/l3vRlt8kty8KKeHni/giphy.gif?cid=5bcd87eeumlqtq22jd83jf8g5hd086kdi3ko4yz07fxyzsqz&rid=giphy.gif&ct=g"
    />
    <GifGridItem
      id="l3vRlt8kty8KKeHhui"
      key="l3vRlt8kty8KKeHhui"
      title="one punch GIF by Adult Swim"
      url="https://media0.giphy.com/media/l3vRlt8kty8KKeHni/giphy.gif?cid=5bcd87eeumlqtq22jd83jf8g5hd086kdi3ko4yz07fxyzsqz&rid=giphy.gif&ct=g"
    />
  </div>
</Fragment>
`;
